### ISSUE TITLE
request adoption of https://code.dlang.org/packages/decimal

### ISSUE THREAD

**Issue by mw66** [dlang-community/discussions#43]
I found this is the most feature complete dub package dealing with decimal data type.

Unfortunately, these package right now has compilation and unittest errors:

https://github.com/rumbu13/decimal/issues

I think it's worth adoption, and should be maintained. And we have volunteer already:

https://github.com/rumbu13/decimal/issues/9#issuecomment-632843049
```
I would be willing to take over and fix the package.
I have forked it already months ago.
The most important think would be to give me access on code.dlang.org, so users would find it.
```

**Comment by mw66** [comment]
@m3m0ry FYI.

**Comment by m3m0ry** [comment]
I have looked into the repository once again. It has some problems with the testing. But nothing I could not handle.

However the biggest issue is, that is has ieee pdfs in the repo (look into doc), which is under copyright (they sell these standard). It is thus in the git history. So I need to rewrite it's history a bit.

**Comment by wilzbach** [comment]
Hi @mingwugmail and @m3m0ry. I'm sorry for the late reply. I will try to respond faster next time.

tl;dr: I think this package is an ideal candidate for the dlang-community as it well used and we already have three people interested in helping out (myself included).

I sent out invites to the dlang-community organization to both of you. Happy to have you around ;-)

Here's a quick summary of what I think we need to do:

1) Contact @rumbu13 regarding dub package. If we can't reach him, there's always the option of `decimal2` or similar. @mingwugmail have you already tried this? (I'm referring to traditional e-mail)

2) Rewrite history to exclude copyrighted material. @m3m0ry maybe you can look into this?
I have already created https://github.com/dlang-community/decimal, so you could push to this.
Q: Depending on (1) it might be possible to transfer the original repository, but if that fails should we use GH's fork option to mark it better as fork instead?

3) Add decimal to buildkite (e.g. https://github.com/dlang/ci/blob/master/buildkite.sh#L110). This way we can avoid it from breaking again. I can/will look into this.

**Comment by m3m0ry** [comment]
Hi @wilzbach . Thank you for your answer.

1. I will contact @rumbu13 regarding the dub package, once we made some progress.

2. I think I have already removed the copyright material on my fork. I will double check it to confirm, that it is not somewhere in the git.
We will wait how @rumbu13 answers. Let me make a PR (4). If he reacts, we could ask him to transfer. If he does not react, forking is then pointless? (A link to original repository would suffice).
 
3. I don't know `buildkite` and I would appreciate help with it. (Depends on (4))

4. I will look into the tests of `decimal` tonight. Getting this package to run it's test is the first thing we need to do. I have looked into it in the past, but I didn't have the motivation.

5. After having a working package, we can the discuss further approach (more tests, IEEE tests, benchmarking, dub package, moving all of this into the proper repository).

**Comment by m3m0ry** [comment]
4. I have repaired `test2.d` (at least i think so). I have some progress on `test.d`, however there are still some problems.

**Comment by m3m0ry** [comment]
2. I rewrote history. PDF files are no longer there.

**Comment by m3m0ry** [comment]
4. I have "repaired" `test.d` (i have commented some test cases). However `test.d` generates `intel.d` which is still not compilable.

1. I will contact @rumbu13 , in order to provide us with some info.

**Comment by m3m0ry** [comment]
Moving this issue to #https://github.com/dlang-community/decimal/issues/1

Please close.
